### PR TITLE
GitHub Actions working on Mac build

### PR DESCRIPTION
**Pull request by brtietz**
Add Mac runner to GitHub actions.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,18 +7,72 @@ on:
...
+    - name: Test
+      # Turn off fast fail for when the landbosse tests write to cerr
+      shell: bash
+      run: |
+        set -e
+        ${SSCDIR}/build/test/Test
+
+      
+  build-on-mac:
+    runs-on: macos-latest
```
Comment (sjanzou) [inline .github/workflows/ci.yml:65]: Do you know the mac runner hardware and macOS version? what is macos-latest?

**Review by sjanzou — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (brtietz) [reply .github/workflows/ci.yml:65]: `macOS Catalina 10.15`, as per https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners

Comment (brtietz) [reply .github/workflows/ci.yml:65]: Apparently they're in the process of migrating: https://github.com/actions/virtual-environments/issues/4060
It'd probably safer to fix a version here, let me know which is preferred.

Comment (sjanzou) [reply .github/workflows/ci.yml:65]: For reference, the 2021.12.02 release was built using 11.6.1 on an Intel based Macbook Pro.
The macOS_wx3.1.5 was built using 12.0.1 on an Apple Silicon M1 based Macbook Air for Intel and Apple Silicon (using Rosetta) based Macs.

**Pull request merged**